src/EventGridExtension/TriggerBinding/EventGridTriggerAttributeBindingProvider.cs

```diff
@@ -64,13 +67,28 @@ public EventGridTriggerBinding(ParameterInfo parameter, EventGridExtensionConfig
 
             public Type TriggerValueType
             {
-                get { return typeof(JObject); }
+                get { return _singleDipatch ? typeof(JObject) : typeof(JArray); }
             }
 
             // Extract binding data
             // Conversion from value to parameterType is done by GenericCompositeBindingProvider
             public Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
             {
+                if (!_singleDipatch)
+                {
+                    if (!(value is JArray triggerVals))
+                    {
+                        throw new InvalidOperationException($"Unable to bind {value} to type {_parameter.ParameterType}. " +
+                            $"Expected {nameof(value)} to be of type {typeof(JArray)}");
+                    }
+                    var eventDataCollection = triggerVals.Select(ev => ev["data"]);
+                    var bindingDataCollection = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase)
+                    {
+                        { "data", eventDataCollection }
```

Comment:
I don't think this should have any implications, but would love to verify. I am not entirely sure when is this binding data ever used. All the tests passed, including my manual tests. I followed the request in local testing, and didn't see this `"data"` every being actually used by the host. I think the logger might be using it somehow. But I will do more tests in Azure and then maybe I will see logs involving this.

@fabiocav , fyi

Reply:
The values in the binding data collection support method signatures binding directly to these values. E.g. in this case, your function could have an `object data`  param and it will be bound to this value.

In the case of this extension, I'm not sure what value if any this binding data is actually adding, since when binding to JObject/EventGridEvent you already have this data on the trigger value itself. An example of a case where binding data IS useful is the queue trigger binding, where you might want to bind the message to a string param, but also have a param that binds to the dequeueCount, etc. See https://github.com/Azure/azure-webjobs-sdk/blob/d4b3cc247c950b44e60a5f4c712c8bfafd1d4653/src/Microsoft.Azure.WebJobs.Extensions.Storage/Queues/Triggers/QueueTriggerBinding.cs#L153

Anyhow, you've followed the existing pattern, so it works for an array of events as well. In that case user would bind to the data via `object[] data`

Reply:
Thanks @mathewc, that makes a lot of sense. I will add these test. Looks like there's already a couple in `ValidTriggerDataResolveTests`. Will update the PR with more bacth tests.

Reply:
@mathewc, I added the tests for these scenarios.

Instead of,
```
public void TestBindingData_Batch([EventGridTrigger] JArray value, object[] data)
```
I had to go with, 
```
public void TestBindingData_Batch([EventGridTrigger] JObject[] value, object[] data)
```
because `JArray` is not an array type.


Also if user has,
```
public void Foo([EventGridTrigger] JObject value, object data)
```
and they receive a batch of events, it will be unable to bind to data, which I assume is reasonable.